README.md

```diff
@@ -6,14 +6,140 @@ A validator for CSV files adhering to the [Data Models](https://github.com/chop-
 
 ## Usage
 
 Run the following to see the usage.
 
 ```
-data-models-validator -help
+$ data-models-validator -help
+Data Models Validator - 0.1.0-beta.1
+
+Usage:
+
+  data-models-validator [-model <model>]
+                        [-version <version>]
+                        [-delim <delimiter>]
+                        [-compr <compression>]
+                        [-service <service>]
+                        ( <file>[:<table>]... | [:<table>] )
+
+The Data Models Validator reads a file containing data and checks it against
+the data model's schema. Input files or stream are delimited files (such as CSV)
+and optionally compressed using gzip or bzip2.
+
+One or more existing input files can be explicitly passed otherwise STDIN
+will be read. Each input can be optionally annotated with an explicit table name
+to be validated against. If not specified, the file name will be used to
+determine which table the file corresponds to.
+
+Source: https://github.com/chop-dbhi/data-models-validator
+
+Examples:
+
+  # Validate person.csv file against the OMOP v5 data model.
+  data-models-validator -model omop -version 5.0.0 person.csv
+
+  # Validate foo.csv against the person table in the OMOP v5 data model.
+  data-models-validator -model omop -version 5.0.0 foo.csv:person
+
+  # Validate the STDIN stream denoting it is tab-delimited and gzipped.
+  data-models-validator -model omop -version 5.0.0 -delim $'\t' -compr gzip
+
+```
+
+## Functionality
+
+The validator checks the following:
+
+- header matches fields of specified table
+- each row of data has the correct number of fields
+- data is encoded in UTF8
+- quotes within data values are escaped
+- date and datetime data is valid and properly formatted
+- integer and number (float) data is valid and fits in 32-bit types
+- required data is not left null
+- string data does not exceed defined max lengths
+
+The validator does **not** check:
+
+- foreign key referential integrity
+- data model conventions such as correct concept usage
+- uniqueness of data across rows
+
+## Known Bugs
+
+If the validator is run several times in quick succession, an error from the underlying data models service is thrown:
+
+```
+$ data-models-validator -model i2b2_pedsnet -version 2.0.0 obs-fact.csv:observation_fact
+error decoding model revisions: invalid character '<' looking for beginning of value
+```
+
+No harm is done to any data and by waiting several minutes before re-running, the problem can be circumvented. The development team is working on a fix ASAP and you can track our progress [here](https://github.com/chop-dbhi/data-models-validator/issues/9)
+
+## Future Directions
+
+Soon, we hope to add higher level validation checks, especially of referential integrity.
+
+## Output Examples
+
+If the filename does not match a table in the model, a list of known tables in the model is printed. The fix is to specify the table by adding `:table_name` to the end of your file name:
```

Comment:
`The fix` &rarr; `The solution` sounds more appropriate since this was an intentional design.